BaragonService/src/main/java/com/hubspot/baragon/service/elb/ApplicationLoadBalancer.java

```diff
@@ -29,23 +62,401 @@ public ApplicationLoadBalancer(Optional<ElbConfiguration> configuration,
+      }
+    }
+  }
+
+  private boolean agentShouldRegisterInTargetGroup(String baragonAgentInstanceId, TargetGroup targetGroup) {
+    DescribeTargetHealthRequest targetHealthRequest = new DescribeTargetHealthRequest()
+        .withTargetGroupArn(targetGroup.getTargetGroupArn());
+    List<TargetHealthDescription> targets = elbClient.describeTargetHealth(targetHealthRequest)
+        .getTargetHealthDescriptions();
+
+    boolean shouldRegister = true;
+    for (TargetHealthDescription targetHealthDescription : targets) {
+      if (targetHealthDescription.getTarget().getId().equals(baragonAgentInstanceId)) {
+        shouldRegister = false;
+      }
+    }
+    return shouldRegister;
+  }
+
+  private Collection<String> getSubnetsFromLoadBalancer(LoadBalancer loadBalancer) {
+    List<AvailabilityZone> availabilityZones = loadBalancer.getAvailabilityZones();
+    Set<String> subnetIds = new HashSet<>();
+
+    for (AvailabilityZone availabilityZone : availabilityZones) {
+      subnetIds.add(availabilityZone.getSubnetId());
+    }
+
+    return subnetIds;
+  }
+
+  private Optional<TargetGroup> guaranteeTargetGroupFor(BaragonGroup baragonGroup) {
+    DescribeTargetGroupsRequest targetGroupsRequest = new DescribeTargetGroupsRequest()
+        .withNames(baragonGroup.getName());
+    List<TargetGroup> targetGroups = elbClient.describeTargetGroups(targetGroupsRequest)
+        .getTargetGroups();
+    if (targetGroups.isEmpty()) {
+      LOG.info("No target group set up for BaragonGroup {}. Skipping.", baragonGroup);
+      return Optional.absent();
+    } else {
+      return Optional.of(targetGroups.get(0));
+    }
+  }
+
+  private Collection<TargetDescription> targetsInTargetGroup(TargetGroup targetGroup) {
+    DescribeTargetHealthRequest targetHealthRequest = new DescribeTargetHealthRequest()
+        .withTargetGroupArn(targetGroup.getTargetGroupArn());
+    List<TargetHealthDescription> targetGroupsResult = elbClient
+        .describeTargetHealth(targetHealthRequest)
+        .getTargetHealthDescriptions();
+
+    Collection<TargetDescription> targetDescriptions = new HashSet<>();
+    for (TargetHealthDescription targetHealthDescription : targetGroupsResult) {
+      targetDescriptions.add(targetHealthDescription.getTarget());
+    }
+
+    return targetDescriptions;
+  }
+
+  private Collection<TargetDescription> listOldTargets(BaragonGroup baragonGroup,
+                                                       Collection<TargetDescription> targetsOnGroup,
+                                                       Collection<BaragonAgentMetadata> knownAgents) {
+    Collection<String> knownAgentIds = agentIds(knownAgents);
+
+    Collection<TargetDescription> removableTargets = new HashSet<>();
+    for (TargetDescription targetDescription : targetsOnGroup) {
+      if (! knownAgentIds.contains(targetDescription.getId()) && canDeregisterAgent(baragonGroup, targetDescription.getId())) {
```

Comment:
you should be able to copy some of the logic from [here](https://github.com/HubSpot/Baragon/pull/202/files#diff-e632d313cd41e36b4dafd51ba5dd83a2R283) for this. If you find yourself rewriting bits of those methods, feel free to move them to the `ElasticLoadBalancer` class so they can be shared